src/main/java/org/apache/commons/lang3/Conversion.java

```diff
@@ -0,0 +1,1563 @@
+     * <p>
+     * Converts an array of int into a long using the default (little endian, Lsb0) byte and bit
+     * ordering.
+     * </p>
+     *
+     * @param src the int array to convert
+     * @param srcPos the position in {@code src}, in int unit, from where to start the
+     *            conversion
+     * @param dstInit initial value of the destination long
+     * @param dstPos the position of the lsb, in bits, in the result long
+     * @param nInts the number of ints to convert
+     * @return a long containing the selected bits
+     * @throws IllegalArgumentException if {@code (nInts-1)*32+dstPos >= 64}
+     * @throws NullPointerException if {@code src} is {@code null}
+     * @throws ArrayIndexOutOfBoundsException if {@code srcPos + nInts > src.length}
+     */
+    public static long intArrayToLong(final int[] src, final int srcPos, final long dstInit, final int dstPos,
+            final int nInts) {
+        if (src.length == 0 && srcPos == 0 || 0 == nInts) {
+            return dstInit;
+        }
+        if ((nInts - 1) * 32 + dstPos >= 64) {
+            throw new IllegalArgumentException("(nInts-1)*32+dstPos is greater or equal to than 64");
+        }
+        long out = dstInit;
+        for (int i = 0; i < nInts; i++) {
+            final int shift = i * 32 + dstPos;
+            final long bits = (0xffffffffL & src[i + srcPos]) << shift;
+            final long mask = 0xffffffffL << shift;
+            out = (out & ~mask) | bits;
+        }
+        return out;
+    }
+
+    /**
+     * <p>
+     * Converts an array of short into a long using the default (little endian, Lsb0) byte and
+     * bit ordering.
+     * </p>
+     *
+     * @param src the short array to convert
+     * @param srcPos the position in {@code src}, in short unit, from where to start the
+     *            conversion
+     * @param dstInit initial value of the destination long
+     * @param dstPos the position of the lsb, in bits, in the result long
+     * @param nShorts the number of shorts to convert
+     * @return a long containing the selected bits
+     * @throws NullPointerException if {@code src} is {@code null}
+     * @throws IllegalArgumentException if {@code (nShorts-1)*16+dstPos >= 64}
+     * @throws ArrayIndexOutOfBoundsException if {@code srcPos + nShorts > src.length}
+     */
+    public static long shortArrayToLong(final short[] src, final int srcPos, final long dstInit, final int dstPos,
+            final int nShorts) {
+        if (src.length == 0 && srcPos == 0 || 0 == nShorts) {
+            return dstInit;
+        }
+        if ((nShorts - 1) * 16 + dstPos >= 64) {
+            throw new IllegalArgumentException("(nShorts-1)*16+dstPos is greater or equal to than 64");
+        }
+        long out = dstInit;
+        for (int i = 0; i < nShorts; i++) {
+            final int shift = i * 16 + dstPos;
+            final long bits = (0xffffL & src[i + srcPos]) << shift;
+            final long mask = 0xffffL << shift;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).